src/openrct2/peep/Guest.cpp

```diff
@@ -1551,12 +1551,12 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
     {
         if (price != 0)
         {
-            if (cash_in_pocket == 0)
+            if ((cash_in_pocket == 0) && !(gParkFlags & PARK_FLAGS_NO_MONEY))
```

Comment:
The park flags check can be moved up to line 1552, rather than checking it twice in lines 1554 and 1559.

Reply:
I've consolidated the check to only be on line 1552 now as you said. I pushed this commit. Hopefully I am doing things correctly.